ground/src/main/java/com/google/android/ground/model/geometry/GeometryValidator.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.model.geometry
+
+object GeometryValidator {
+  private fun List<Any>.isFirstAndLastSame(): Boolean = firstOrNull() == lastOrNull()
+
+  /** Validates that the current [LinearRing] is well-formed. */
+  fun LinearRing.validate() {
```

Comment:
Since we'll probably use variations on this for different types of Geometries, I think we can define this as a function in the Geometry interface. A trivial default implementation would just be the no-op Unit function (a naked `return`).

Reply:
Done